File: examples/spacetrack/filter_pattern.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Demonstrates pattern matching in SpaceTrack queries.
"""

import os

import brahe as bh

# Get credentials from environment
# Note: For production use, most users should use the default constructor:
#   client = bh.SpaceTrackClient(user, password)
# These examples use the test server for automated testing.
user = os.environ.get("TEST_SPACETRACK_USER", "")
password = os.environ.get("TEST_SPACETRACK_PASS", "")

if not user or not password:
    print("Set TEST_SPACETRACK_USER and TEST_SPACETRACK_PASS environment variables")
else:
    client = bh.SpaceTrackClient(
        user, password, base_url="https://for-testing-only.space-track.org/"
    )

    # Like operator: pattern matching with % wildcard
    # Find all Starlink satellites
    starlinks = client.gp(
        object_name="~~STARLINK%",
        limit=10,
    )
    print(f"Found {len(starlinks)} Starlink satellites")

    # Starts with: international designator prefix
    # Find objects from 2024 launches
    launches_2024 = client.satcat(
        intldes="^2024-",
        limit=10,
    )
    print(f"Found {len(launches_2024)} objects from 2024 launches")
    # Found 10 Starlink satellites
    # Found 10 objects from 2024 launches

print("Example completed successfully!")

File: examples/spacetrack/filter_pattern.rs
//! Demonstrates pattern matching in SpaceTrack queries.
//!
//! FLAGS = ["CI-ONLY"]

use brahe::spacetrack::{BlockingSpaceTrackClient, TEST_BASE_URL};

fn main() {
    // Get credentials from environment
    // Note: For production use, most users should use the default constructor:
    //   BlockingSpaceTrackClient::new("user", "password")
    // These examples use the test server for automated testing.
    let user = std::env::var("TEST_SPACETRACK_USER").unwrap_or_default();
    let password = std::env::var("TEST_SPACETRACK_PASS").unwrap_or_default();

    if user.is_empty() || password.is_empty() {
        println!("Set TEST_SPACETRACK_USER and TEST_SPACETRACK_PASS environment variables");
    } else {
        let client = BlockingSpaceTrackClient::with_base_url(&user, &password, TEST_BASE_URL)
            .expect("Failed to create client");

        // Pattern matching with LIKE operator (~~)
        // Use %% for wildcard matching
        let starlink_records = client
            .gp(
                None,
                Some("~~STARLINK%"), // Names starting with STARLINK
                None,
                None,
                None,
                None,
                Some(5),
                None,
            )
            .expect("Query failed");

        println!("Found {} STARLINK satellites", starlink_records.len());

        // Starts with operator (^)
        let iss_records = client
            .gp(
                None,
                Some("^ISS"), // Names starting with ISS
                None,
                None,
                None,
                None,
                Some(5),
                None,
            )
            .expect("Query failed");

        println!("Found {} ISS-related objects", iss_records.len());
        // Found 5 STARLINK satellites
        // Found 5 ISS-related objects
    }

    println!("Example completed successfully!");
}
